stable_baselines3/common/evaluation.py

```diff
@@ -49,15 +58,32 @@ def evaluate_policy(
         episode_length = 0
         while not done:
             action, state = model.predict(obs, state=state, deterministic=deterministic)
-            obs, reward, done, _info = env.step(action)
+            obs, reward, done, info = env.step(action)
             episode_reward += reward
             if callback is not None:
                 callback(locals(), globals())
             episode_length += 1
             if render:
                 env.render()
-        episode_rewards.append(episode_reward)
-        episode_lengths.append(episode_length)
+
+        # Remove VecEnv stacking (if any)
+        if isinstance(env, VecEnv):
+            info = info[0]
+
+        if "episode" in info.keys():
```

Comment:
did you try that on atari games?
I'm afraid it will have some weird behavior because the wrapper makes "end of life = end of episode", so you will have a mix of normalized and unnormalized reward, no?

One solution would be to deactivate that behavior when testing atari games (can be done in the zoo).

Reply:
Ah right, I felt like I forgot something when cooking this up... Good thing you mentioned this before I started running my Atari experiments where I use this ^^.

I would avoid Atari-specific tweaking and just go straight for bit more generic solution. I think a reliable check for ``Monitor`` wrapper would do the trick (need to think more on this).